tools/RAiDER/downloadGNSSDelays.py

```diff
@@ -0,0 +1,330 @@
+#!/usr/bin/env python3
+import os
+import sys
+import requests
+import argparse
+import itertools
+import pandas as pd
+from RAiDER.getStationDelays import getStationData
+
+
+#base URL for UNR repository
+_UNR_URL = "http://geodesy.unr.edu/"
+
+
+def download_UNR(statID, year, writeDir = '.', baseURL = _UNR_URL, download = False, verbose = False):
+    '''
+    Download a zip file containing tropospheric delays for a given station and year
+    The URL format is http://geodesy.unr.edu/gps_timeseries/trop/<ssss>/<ssss>.<yyyy>.trop.zip
+    Inputs:
+        statID   - 4-character station identifier
+        year     - 4-numeral year
+    '''
+    URL = "{0}gps_timeseries/trop/{1}/{1}.{2}.trop.zip".format(baseURL, statID.upper(), year)
+    if download:
+        saveLoc = os.path.abspath(os.path.join(writeDir, '{0}.{1}.trop.zip'.format(statID.upper(), year)))
+        filepath = download_url(URL, saveLoc, verbose=verbose)
+    else:
+        filepath = check_url(URL, verbose=verbose)
+    return filepath
+
+
+def getStatsByllh(llhBox = None, baseURL = _UNR_URL, userstatList = None):
+    '''
+    Function to pull lat, lon, height, beginning date, end date, and number of solutions for stations inside the bounding box llhBox. 
+    llhBox should be a tuple with format (lat1, lat2, lon1, lon2), where lat1, lon1 define the lower left-hand corner and lat2, lon2 
+    define the upper right corner. 
+    '''
+    import pandas as pd
+    from urllib.request import urlopen 
```

Comment:
Yeah, pandas was redundant here anyways.

And I don't believe there was a particular reason for choosing ```urlib``` over ```requests```. Any reason to proceed with the latter though @Askaholic?

Reply:
Mostly because `requests` is a higher level library. It will be able to handle stuff like redirects, cookies, etc out of the box. Even the documentation of `urllib` recommends using `requests` instead https://docs.python.org/3/library/urllib.request.html#module-urllib.request